be/src/storage/tablet_schema.cpp

```diff
@@ -316,6 +318,101 @@ void TabletColumn::add_sub_column(TabletColumn&& sub_column) {
     _get_or_alloc_extra_fields()->sub_columns.emplace_back(std::move(sub_column));
 }
 
+/******************************************************************
+ * TabletIndex
+ ******************************************************************/
+
+Status TabletIndex::init_from_thrift(const TOlapTableIndex& index, const TabletSchema& tablet_schema) {
+    _index_name = index.index_name;
+    // init col_unique_id in index at be side, since col_unique_id may be -1 at fe side
+    // get column unique id by name
+    std::vector<int32_t> col_unique_ids(index.columns.size());
+    for (size_t i = 0; i < index.columns.size(); i++) {
+        auto column_idx = tablet_schema.field_index(index.columns[i]);
+        if (column_idx >= 0) {
+            col_unique_ids[i] = tablet_schema.column(column_idx).unique_id();
+        } else {
+            return Status::Cancelled("Can not find column " + index.columns[i] + " in GIN index. ");
+        }
+    }
+    _col_unique_ids = std::move(col_unique_ids);
+
+    switch (index.index_type) {
```

Comment:
This can be extracted to be a separated function that can convert from thrift Index Type to IndexType.